src/views/projects/Issues.svelte

```diff
@@ -1,5 +1,7 @@
 <script lang="ts" context="module">
-  export type State = "open" | "closed";
+  import type { IssueState } from "@app/lib/issue";
+
+  export type State = Extract<IssueState["status"], "open" | "closed">;
```

Comment:
The `Extract` doesn't seem to give us anything here, how about:

```diff
diff --git a/src/views/projects/Issues.svelte b/src/views/projects/Issues.svelte
index db9bf881..f6c9ce3e 100644
--- a/src/views/projects/Issues.svelte
+++ b/src/views/projects/Issues.svelte
@@ -1,7 +1,7 @@
 <script lang="ts" context="module">
   import type { IssueState } from "@app/lib/issue";
 
-  export type State = Extract<IssueState["status"], "open" | "closed">;
+  export type IssueStatus = IssueState["status"];
 </script>
 
 <script lang="ts">
@@ -21,22 +21,22 @@
   import TabBar from "@app/components/TabBar.svelte";
 
   export let issues: Issue[];
-  export let state: State;
+  export let status: IssueStatus;
   export let project: Project;
 
-  let options: Tab<State>[];
+  let options: Tab<IssueStatus>[];
 
-  const stateOptions: State[] = ["open", "closed"];
+  const stateOptions: IssueStatus[] = ["open", "closed"];
   $: options = stateOptions.map<{
-    value: State;
+    value: IssueStatus;
     title: string;
     disabled: boolean;
-  }>((s: State) => ({
+  }>((s: IssueStatus) => ({
     value: s,
     title: `${project.issues[s]} ${s}`,
     disabled: project.issues[s] === 0,
   }));
-  $: filteredIssues = groupIssues(issues)[state];
+  $: filteredIssues = groupIssues(issues)[status];
   $: sortedIssues = filteredIssues.sort(
     ({ timestamp: t1 }, { timestamp: t2 }) => t2 - t1,
   );
@@ -77,7 +77,7 @@
           router.updateProjectRoute({
             search: `state=${e.detail}`,
           })}
-        active={state} />
+        active={status} />
     </div>
     <HeaderToggleLabel
       disabled={!$sessionStore || !utils.isLocal(project.seed.host)}
@@ -114,7 +114,7 @@
     </div>
   {:else}
     <Placeholder emoji="🍂">
-      <div slot="title">{capitalize(state)} issues</div>
+      <div slot="title">{capitalize(status)} issues</div>
       <div slot="body">No issues matched the current filter</div>
     </Placeholder>
   {/if}
diff --git a/src/views/projects/Patches.svelte b/src/views/projects/Patches.svelte
index fcfcc1c4..b417f42e 100644
--- a/src/views/projects/Patches.svelte
+++ b/src/views/projects/Patches.svelte
@@ -1,8 +1,5 @@
 <script lang="ts" context="module">
-  export type State = Extract<
-    PatchState["status"],
-    "draft" | "proposed" | "archived"
-  >;
+  export type PatchStatus = PatchState["status"];
 </script>
 
 <script lang="ts">
@@ -15,7 +12,7 @@
   import Placeholder from "@app/components/Placeholder.svelte";
 
   export let patches: Patch[];
-  export let state: State;
+  export let status: PatchStatus;
   export let project: Project;
 </script>
 
@@ -60,7 +57,7 @@
     </div>
   {:else}
     <Placeholder emoji="🍂">
-      <div slot="title">{utils.capitalize(state)} patches</div>
+      <div slot="title">{utils.capitalize(status)} patches</div>
       <div slot="body">No issues matched the current filter</div>
     </Placeholder>
   {/if}
diff --git a/src/views/projects/View.svelte b/src/views/projects/View.svelte
index 60792b68..66b3cdb5 100644
--- a/src/views/projects/View.svelte
+++ b/src/views/projects/View.svelte
@@ -1,7 +1,7 @@
 <script lang="ts">
   import type { ProjectRoute } from "@app/lib/router/definitions";
-  import type { State as IssueState } from "./Issues.svelte";
-  import type { State as PatchState } from "./Patches.svelte";
+  import type { IssueStatus } from "./Issues.svelte";
+  import type { PatchStatus } from "./Patches.svelte";
 
   import * as issue from "@app/lib/issue";
   import * as patch from "@app/lib/patch";
@@ -32,10 +32,10 @@
   $: seed = activeRoute.params.seed;
 
   $: searchParams = new URLSearchParams(activeRoute.params.search || "");
-  $: issueFilter = (searchParams.get("state") as IssueState) || "open";
+  $: issueFilter = (searchParams.get("state") as IssueStatus) || "open";
   $: patchTabFilter =
     (searchParams.get("tab") as "activity" | "commits") || "activity";
-  $: patchFilter = (searchParams.get("state") as PatchState) || "proposed";
+  $: patchFilter = (searchParams.get("state") as PatchStatus) || "proposed";
 
   const getProject = async (id: string, seed: string, peer?: string) => {
     const project = await proj.Project.get(id, seed, peer);
@@ -173,7 +173,7 @@
         {#await issue.Issue.getIssues(project.id, project.seed.addr)}
           <Loading center />
         {:then issues}
-          <Issues {project} state={issueFilter} {issues} />
+          <Issues {project} status={issueFilter} {issues} />
         {:catch e}
           <div class="message">
             <Message error>{e.message}</Message>
@@ -193,7 +193,7 @@
         {#await patch.Patch.getPatches(project.id, project.seed.addr)}
           <Loading center />
         {:then patches}
-          <Patches {patches} state={patchFilter} {project} />
+          <Patches {patches} status={patchFilter} {project} />
         {:catch e}
           <div class="message">
             <Message error>{e.message}</Message>

```

Reply:
235aee69d42ddf3d8384c295946d858d451a6196